elastic/kibana
# Total number of alerts, broken down by alerting rule types telemetry

Resolves: https://github.com/elastic/response-ops-team/issues/150

This PR adds a new telemetry to the alerting plugin.
The task collects the below data and saves in the task state.

`count_alerts_total` total alerts of all time
`count_alerts_by_rule_type`  total alerts of all time by rule types

Note: I tried to use the stats API as Brandon suggested in the issue, but it just returns the total number of alerts.
We have to use aggregations for alerts by rule types. Therefore I didn't use it.

## To verify:


- Change the task interval to 1m on [this line](https://github.com/elastic/kibana/pull/192866/files#diff-014c1a7c63ade0d0f548523ef161369fdeb21c589a8112f202b8086ca23af6fdL32)
- Run Kibana
- Create some rules that generates alerts.
- Let them run for a while.
- Check the saved alerting telemetry in the task state by using the below query.
- `count_alerts_total`and `count_alerts_by_rule_type` should be populated

```
GET /.kibana_task_manager_*/_search
{
    "query": {
      "term": {
        "task.taskType": "alerting_telemetry"        
      }
    },
    "size" : 1
}
```
